pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -403,16 +403,20 @@ async function resolveDependenciesOfImporters (
   ctx: ResolutionContext,
   importers: ImporterToResolve[]
 ): Promise<ResolveDependenciesOfImportersResult> {
-  const extendedWantedDepsByImporters = importers.map(({ wantedDependencies, options }) => getDepsToResolve(wantedDependencies, ctx.wantedLockfile, {
-    preferredDependencies: options.preferredDependencies,
-    prefix: options.prefix,
-    proceed: options.proceed || ctx.forceFullResolution,
-    registries: ctx.registries,
-    resolvedDependencies: options.resolvedDependencies,
-  }))
+  const extendedWantedDepsForImporterPairs = importers.map((importer): [ExtendedWantedDependency[], ImporterToResolve] => {
+    const { wantedDependencies, options } = importer
+    const extendedWantedDeps = getDepsToResolve(wantedDependencies, ctx.wantedLockfile, {
+      preferredDependencies: options.preferredDependencies,
+      prefix: options.prefix,
+      proceed: options.proceed || ctx.forceFullResolution,
+      registries: ctx.registries,
+      resolvedDependencies: options.resolvedDependencies,
+    })
+    return [extendedWantedDeps, importer]
```

Comment:
it can be changed to an object.

Reply:
Yeah, this could also be a map of `Map<Importer, ExtendedWantedDeps[]>`. Would you prefer that over an array of pairs?

Reply:
I'm recalling the reason I didn't go with a `Map`. The [`Iterator.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/map) method isn't widely implemented yet, so you end up converting the map back into a pairs array. 😞

```ts
await Promise.all([...extendedWantedDepsByImporter.entries()]
  .map(async ([importer, extendedWantedDeps]) => {
    // ...
  })
)
```

Reply:
an object is fine

Reply:
The object would need to be keyed by a `string`. Were you thinking of copying the `id` field to `ImporterToResolve`?

https://github.com/pnpm/pnpm/blob/73b241d27b37a39ac32e7b7999f039701a11c0ea/pkg-manager/resolve-dependencies/src/resolveDependencies.ts#L388-L394